src/java/org/apache/cassandra/index/Index.java

```diff
@@ -644,9 +646,18 @@ default boolean supportsReplicaFilteringProtection(RowFilter rowFilter)
 
         /**
          * @param executionController the collection of OpOrder.Groups which the ReadCommand is being performed under.
+         * @param tombstonesToSkip
          * @return partitions from the base table matching the criteria of the search.
          */
-        public UnfilteredPartitionIterator search(ReadExecutionController executionController);
+        public UnfilteredPartitionIterator search(ReadExecutionController executionController, Set<PrimaryKey> tombstonesToSkip);
+
+        /**
+         * @return true if the query being searched involves an ORDER BY clause
+         */
+        default Expression getOrderBy()
```

Comment:
`Expression` is a SAI class, should not be part of `Index` interface. It's only used to get `IndexContext` which is available via `StorageAttachedIndexQueryPlan`